course04/homework00/ciaranfrain/sorted_array.cpp

```diff
@@ -0,0 +1,30 @@
+#include "sorted_array.h"
+
+#include <algorithm>
+#include <stdexcept>
+
+SortedArray::SortedArray(std::size_t maxSize): mMaxSize(maxSize)
+{}
+
+void SortedArray::Insert(int n)
+{
+	if (size() == mMaxSize)
+		throw std::length_error("error: sorted array is at maximum capacity");
+
+	if (empty())
+		mStorage.push_back(n);
+	else if (n < mStorage.front())
+		mStorage.push_front(n);
+	else if (n > mStorage.back())
+		mStorage.push_back(n);
```

Comment:
I see why you used deque now - makes sense. Here I would personally stick to one of those two options:
1. simplicity: push_back + sort
2. binary search + insert

that being said, the rule before doing 2 is: you have to measure. if you dont, it might be slower than 1.